patricia.c

```diff
@@ -94,7 +94,7 @@ comp_with_mask (void *addr, void *dest, u_int mask)
 
     if ( /* mask/8 == 0 || */ memcmp (addr, dest, mask / 8) == 0) {
 	int n = mask / 8;
-	int m = ((-1) << (8 - (mask % 8)));
+	int m = -(-1 << (8 - (mask % 8)));
```

Comment:
Isn't this still left-shifting a negative value?

Reply:
It totally was, thanks.  Meant it to be the same change as in Zeek: https://github.com/zeek/zeek/pull/1174/commits/428870be15bfb8ef3a571a9b41cf84ace51ec6bf